simplistic_editor/lib/basic_text_input_client.dart

```diff
@@ -315,22 +315,34 @@ class BasicTextInputClientState extends State<BasicTextInputClient>
     PasteTextIntent: CallbackAction<PasteTextIntent>(
       onInvoke: (intent) => pasteText(intent.cause),
     ),
+    DoNothingAndStopPropagationTextIntent: DoNothingAction(
+      consumesKey: false,
+    ),
   };
 
-  void _delete() {
+  void _delete(bool forward) {
     if (_value.text.isEmpty) return;
 
     late final TextRange deletedRange;
     late final TextRange newComposing;
-    final int deletedLength =
-        _value.text.substring(0, _selection.baseOffset).characters.last.length;
+    late final String deletedText;
+    final int offset = _selection.baseOffset;
 
     if (_selection.isCollapsed) {
-      if (_selection.baseOffset == 0) return;
-      deletedRange = TextRange(
-        start: _selection.baseOffset - deletedLength,
-        end: _selection.baseOffset,
-      );
+      if (forward) {
+        deletedText = _value.text.substring(offset).characters.first;
```

Comment:
Maybe we also need
`if (_selection.baseOffset == _value.text.length) return;`

Reply:
Good catch!